---
title: "volcano fix"
knit: (function(input_file, encoding) {
  out_dir <- './';
  rmarkdown::render(input_file,
  encoding=encoding,
  output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
output: 
  html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(manhattanly)
```


```{r}
tmm_r <- read.delim(file = "mate_00vs02_tmm_filt_annot.txt", header=TRUE, row.names = NULL, com='', check.names=F)
head(tmm_r)
volc_obj <- volcanor(tmm_r, p = "P", effect_size = "EFFECTSIZE", snp = "locus_tag", 
                     gene = "protein", annotation1 = "old_locus_tag", annotation2 = "arCOG_name")
volc_plot <- volcanoly(volc_obj, effect_size_line = c(-3,3), effect_size_line_color = "orange", 
          genomewideline = -log10(1e-2), genomewideline_color = "green", title = "t00 vs t02")
volc_plot_fixed <- volcanoly(volc_obj, effect_size_line = c(-3,3), effect_size_line_color = "orange", 
          genomewideline = -log10(1e-2), genomewideline_color = "green", title = "t00 vs t02", 
          highlight = tmm_r$locus_tag, highlight_color = "black")
```

```{r, fig.cap="Volcano plot doesn't have annotations for non-significant points", message=FALSE, warning=FALSE, fig.align="center"}
volcanoly(volc_obj, effect_size_line = c(-3,3), effect_size_line_color = "orange", 
          genomewideline = -log10(1e-2), genomewideline_color = "green", title = "t00 vs t02")
```

```{r, fig.cap="Annotations present for all, but significant are no longer separately highlighted", message=FALSE, warning=FALSE, fig.align="center"}
volcanoly(volc_obj, effect_size_line = c(-3,3), effect_size_line_color = "orange", 
          genomewideline = -log10(1e-2), genomewideline_color = "green", title = "t00 vs t02", 
          highlight = tmm_r$locus_tag, highlight_color = "black")
```

```{r testing interactive}
library(ggplot2)
library(plotly)
p <- ggplot(tmm_r) +
  geom_point(data = tmm_r, aes(x = log2FC, y = -log10(P), text = paste(locus_tag,"\n",protein,"\n",arCOG_name,"\n",old_locus_tag)),
             color = "black", cex = 2) +
#  geom_point(data = tmm_r, aes(x = log2FC, y = P), color = "lightgreen", cex = 3) +
  # geom_point(data = dfm1, aes(x = logFC.x, y = logFDR.x), color = "blue", cex = 3) +
  # geom_text(data = dfm1, aes(x = logFC.x, y = logFDR.x, label = gene), hjust = 1, vjust = 2) +
  theme_bw() +
  xlab("log(2) fold change") +
  ylab("log(10) FDR") +
  geom_vline(xintercept = 2, col = "red", linetype = "dashed", size = 0.5) +
  geom_vline(xintercept = -2, col = "red", linetype = "dashed", size = 0.5) +
  geom_hline(yintercept = 0.05, col = "red", linetype = "dashed", size = 0.5)
p
ggplotly(p, tooltip = "text")
```

